[OJS 3.4] Migrate docxConverter plugin from 3.3

quedaporhacer opened the pull request:
This PR updates the docxConverter plugin to be compatible with OJS 3.4.

Key changes:
- Plugin is now fully supported for OJS 3.4.
- Refactored class names for better readability and descriptive naming.

quedaporhacer committed:
Change (docxToJats fork submodule)

quedaporhacer committed:
Merge pull request #1 from sedici/feature/doctojats-fork-submodule

Change (docxToJats fork submodule)

quedaporhacer committed:
Migración parcial: se acomodaron el llamado al handler y los namespaces

quedaporhacer committed:
Migración parcial: Miracion de DOCXConvertedDocument y partes del Handler

quedaporhacer committed:
get authors from publication

quedaporhacer committed:
migration attachSupplementaryFile

quedaporhacer committed:
Changes in file names

quedaporhacer committed:
Merge pull request #2 from sedici/migration/plugin-ojs-3.3-to-3.4

Migration/plugin ojs 3.3 to 3.4

quedaporhacer committed:
Merge pull request #3 from sedici/dev

stable-3_4_0

quedaporhacer committed:
change submodule path

quedaporhacer committed:
submodule restore

GaziYucel commented on DocxToJatsPlugin.php:
```diff
@@ -58,18 +66,22 @@ function getPluginUrl($request) {
+    $op = $args[1];   
+    //error_log('$_SERVER["REQUEST_URI"]: ' . $_SERVER["REQUEST_URI"]);
+    //error_log('$_GET: ' . print_r($_GET, true));
+
```
This is for debugging I assume.

GaziYucel commented on version.xml:
```diff
@@ -13,10 +13,10 @@
+	<date>2021-05-22</date>
+	<lazy-load>1</lazy-load>
+	<class>DocxToJatsPlugin</class>
 </version>
```
These changes and main class name change will require a migration. Please see the entries in the tables `versions` and `plugin_settings` in version 3.3 and 3.4.

GaziYucel commented on DocxToJatsHandler.php:
```diff
@@ -0,0 +1,187 @@
+		$submissionId = $submissionFile->getData('submissionId');
+
+		// Replaced by Santiago
+		//$submission = Services::get('submission')->get($submissionId);
```
Please remove for production.

TomasMessineo committed:
delete comments

TomasMessineo committed:
change plugin name to docxConverter

TomasMessineo committed:
change plugin name to docxConverter

quedaporhacer committed:
Merge pull request #4 from sedici/hotfix/plugin-name-migration-to-docxConverter

Hotfix/plugin name migration to docx converter

TomasMessineo committed:
chore: Upgrade submodule URL to Vitaliy repo

TomasMessineo committed:
chore: Upgrade submodule URL to Vitaliy repo

TomasMessineo committed:
Add migration files and modify version.xml

TomasMessineo committed:
TEMPORAL FIX: adding index.php for correct visaulization in OJS generic plugin's section

TomasMessineo committed:
Add updateDocxConverterPluginName.php

TomasMessineo committed:
updating updateDocxConverterPluginNAme

quedaporhacer committed:
Merge pull request #5 from sedici/migrate/to-3.5

Migrate/to 3.5

quedaporhacer committed:
Merge pull request #6 from sedici/dev

Dev

TomasMessineo commented:
> @TomasMessineo. Can you please merge your `dev` changes into this PR or open a new one. It's much more work to do a review on a branch for both of us. Thanks again.

@GaziYucel We've just merged the latest changes into the stable-3_4_0 branch, so this PR is now up to date with the points we covered. Ready for your review. Thanks!

GaziYucel committed:
- No code changes
- code format according to PSR
- remaining .inc in phpdoc parts
- curly bracket in class / method definition on the next line
- empty double lines removed
- copyright dates updated for files touched by this PR
- added basic phpdoc to methods

GaziYucel committed:
- removed upgrade logic
- git module url changed to original
- unnecessary require_once removed
- moved submodule require to only if plugin is enabled require_once __DIR__ . "/docxToJats/vendor/autoload.php";
- added method return types
- if method return types and parameter types defined, removed from phpdoc
- underscore removed from method names (was in old days used to mark private methods)
- property types added

TomasMessineo committed:
Merge pull request #7 from GaziYucel/stable-3_4_0_sedici

[OJS 3.4] Migrate docxConverter plugin from 3.3 - review fixes

TomasMessineo committed:
Update .gitmodules

TomasMessineo committed:
Fijar submódulo docxToJats en rama stable-v1.0.0

quedaporhacer committed:
Merge branch 'stable-3_4_0' into dev

quedaporhacer committed:
Merge pull request #8 from sedici/dev

Cambio de submodulo

san-fernandez committed:
Fix: actualizar submódulo docxToJats a stable-v1.0.0
